### DISCUSSION TITLE
Add ability to resize devtool panes

### DISCUSSION THREAD

**Discussion by hheavener-kyd** [TanStack/query#6896] [Ideas]
# About
The new version of devtools looks snazzy, but I have a major issue in that I cannot resize panes within the panel. When I select a query in the panel, the  "Query Details" panel has a fixed width and it's too small for me to do anything with. I have to manually edit the width property to expand it so I can actually view the data. An older version of the devtools that I used stuck to 50% width for the "Query Details" panel but the newer version seems to be about half of that.

So can we please include the ability to resize panes within the overall panel, or at least return the details to 50% width? 

To show the problem, I do a lot of my UI work on my vertical monitor and when I use the devtools on that monitor, this is what it ends up looking like:
![image](https://github.com/TanStack/query/assets/61209753/54bb9b67-8809-4247-8c32-8e24d3851555)

It's also worth noting that I would like to be able to resize or collapse the sub-sections within the query details panel also. For instance, have the ability to collapse `Query Details`, `Actions`, `Data Explorer`, and `Query Explorer` and, let's say I have all of them but `Data Explorer` collapsed, then the `Data Explorer` tab would expand vertically to fill the height of the panel. This would be ideal because the main thing I need to use the devtools for is this specific section and, most of the time, it's too small to be comfortably navigated.

**Comment by troywoy** [comment]
This isn't an ideal solution but perhaps you could override a couple styles? Overriding the fixed `flex` size on `.tsqd-queries-container` and `.tsqd-query-details-container` and applying `resize: horizontal` (with something like `overflow: hidden`) could at least give you native resizing.

**Reply by hheavener-kyd** [reply]
@troywoy if this works for you, can you drop the necessary CSS overrides into a snippet that I could reference? I tried your suggestion but it isn't working for me so I'm not sure if your suggestion doesn't work or if I'm doing it wrong.

**Reply by troywoy** [reply]
The problem you likely have is `flex: 1` prevents resize which is how those panels were styled. Undoing that flex means it'll render poorly, but you could at least have resizing I guess?

```css
.tsqd-queries-container,
.tsqd-query-details-container {
  flex: none !important;
}

.tsqd-queries-container {
  overflow: auto;
  resize: horizontal;
}
```

If you toy around with the styles more you could add some width(s) back but like I said, this wasn't an ideal solution.

**Reply by hheavener-kyd** [reply]
I deleted my previous comment. I misunderstood how this would work. I didn't realize it would be the little draggable thing as with text areas at the bottom of the panel. I suppose this will somewhat work for now, however, like you said, it's not an ideal solution. Thank you for the workaround.

**Reply by hheavener-kyd** [reply]
I am still going to leave this open because I think this needs to be implemented in the devtools without having to do this workaround.

**Reply by troywoy** [reply]
Well I haven't been on a small screen and it looks like it's supposed to stack on top of each other on smaller widths. What version are you on?

![image](https://github.com/TanStack/query/assets/1566003/2d533fac-1456-4b35-91b5-791215d0183a)
![image](https://github.com/TanStack/query/assets/1566003/488acd74-f463-4db2-b214-802934479024)


https://codesandbox.io/p/devbox/github/tanstack/query/tree/main/examples/react/basic?embed=1&file=%2Fsrc%2Findex.jsx&theme=light

**Reply by hheavener-kyd** [reply]
I just added a screenshot to the original post for what it looks like on my vertical monitor using `"@tanstack/react-query-devtools": "^5.18.1"`

**Reply by troywoy** [reply]
Can you look at this codesandbox link on that same vertical monitor? From what I see it should be 50/50 but I wonder if you somehow are overriding styles in your app?

**Reply by hheavener-kyd** [reply]
Also, even in the stacked state, the query details are still limited to half the height of the panel, which is a waste. I would like to be able to expand this so I can prioritize viewing the data when I need it. Otherwise, almost half of the screen real estate is empty space for showing query keys that aren't present because I only have one query being used

**Reply by hheavener-kyd** [reply]
The code sandbox does at least stack horizontally, however, sadly, it doesn't address the issue I stated just above ^

**Reply by troywoy** [reply]
@hheavener-kyd, just making you aware of this in case it helps.

https://x.com/aryan__deora/status/1760340170715402467?s=20
